crates/optimism/flashblocks/src/service.rs

```diff
@@ -167,6 +169,27 @@ where
             },
         )))
     }
+
+    fn verify_canon_integrity(&mut self, cx: &mut Context<'_>) -> Option<PendingBlock<N>> {
```

Comment:
can we find a better name for this and some docs?

i feel like it would be easier to read if this just takes in the new tip hash which would then simplify this check

and we keep polling the the receiver in the poll_next fn and then pipe he tip to `fn on_new_tip(num_hash: BlockNumHash) -> Option<P>` for example

Reply:
@mattsse updated